backend/src/test/java/com/redhat/cloud/notifications/recipients/rbac/RbacRecipientUsersProviderTest.java

```diff
@@ -0,0 +1,225 @@
+package com.redhat.cloud.notifications.recipients.rbac;
+
+import com.redhat.cloud.notifications.recipients.User;
+import com.redhat.cloud.notifications.routers.models.Meta;
+import com.redhat.cloud.notifications.routers.models.Page;
+import io.quarkus.cache.CacheInvalidateAll;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.InjectMock;
+import io.smallrye.mutiny.Uni;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+
+@QuarkusTest
+public class RbacRecipientUsersProviderTest {
+
+    private final String accountId = "test-account-id";
+
+    @ConfigProperty(name = "recipient-provider.rbac.elements-per-page")
+    Integer rbacElementsPerPage;
+
+    @InjectMock
+    @RestClient
+    RbacServiceToService rbacServiceToService;
+
+    @Inject
+    RbacRecipientUsersProvider rbacRecipientUsersProvider;
+
+    @Test
+    public void getAllUsers() {
+        int elements = 95;
+        mockGetUsers(elements, false);
+
+        List<User> users = rbacRecipientUsersProvider.getUsers(accountId, false).await().indefinitely();
+        Assertions.assertEquals(elements, users.size());
+        for (int i = 0; i < elements; ++i) {
+            Assertions.assertEquals(String.format("username-%d", i), users.get(i).getUsername());
+        }
+    }
+
+    @Test
+    public void getAllUsersFromDefaultGroup() {
+        RbacGroup defaultGroup = new RbacGroup();
+        defaultGroup.setPlatformDefault(true);
+        defaultGroup.setUuid(UUID.randomUUID());
+
+        int elements = 133;
+
+        mockGetGroup(defaultGroup);
+        mockGetUsers(elements, false);
+
+        List<User> users = rbacRecipientUsersProvider.getGroupUsers(accountId, false, defaultGroup.getUuid()).await().indefinitely();
+        Assertions.assertEquals(elements, users.size());
+        for (int i = 0; i < elements; ++i) {
+            Assertions.assertEquals(String.format("username-%d", i), users.get(i).getUsername());
+        }
+    }
+
+    @Test
+    public void getAllUsersFromGroup() {
```

Comment:
`getAllGroupsCache()` already does everything that is in this test. This one can be removed.